libraries/extensions/ros2-bridge/python/src/typed/serialize.rs

```diff
@@ -31,81 +34,154 @@ impl serde::Serialize for TypedValue<'_> {
+                        s.serialize_element(&value)?;
+                    }
+                    s.end()
+                }
             }
             DataType::Int64 => {
-                let int_array: Int64Array = self.value.clone().into();
-                let number = int_array.value(0);
-                serializer.serialize_i64(number)
+                let array: &Int64Array = self.value.as_primitive();
+                if array.len() == 1 {
+                    let number = array.value(0);
+                    serializer.serialize_i64(number)
+                } else {
+                    let mut s = serializer.serialize_seq(Some(array.len()))?;
+                    for value in array.iter() {
+                        s.serialize_element(&value)?;
+                    }
+                    s.end()
+                }
             }
             DataType::Float32 => {
-                let int_array: Float32Array = self.value.clone().into();
-                let number = int_array.value(0);
-                serializer.serialize_f32(number)
+                let array: &Float32Array = self.value.as_primitive();
+                if array.len() == 1 {
+                    let number = array.value(0);
+                    serializer.serialize_f32(number)
+                } else {
+                    let mut s = serializer.serialize_seq(Some(array.len()))?;
+                    for value in array.iter() {
+                        s.serialize_element(&value)?;
+                    }
+                    s.end()
+                }
             }
             DataType::Float64 => {
-                let int_array: Float64Array = self.value.clone().into();
-                let number = int_array.value(0);
-                serializer.serialize_f64(number)
+                let array: &Float64Array = self.value.as_primitive();
+                if array.len() == 1 {
+                    let number = array.value(0);
+                    serializer.serialize_f64(number)
+                } else {
+                    let mut s = serializer.serialize_seq(Some(array.len()))?;
+                    for value in array.iter() {
+                        s.serialize_element(&value)?;
+                    }
+                    s.end()
+                }
             }
             DataType::Utf8 => {
-                let int_array: StringArray = self.value.clone().into();
+                let int_array: &StringArray = self.value.as_string();
                 let string = int_array.value(0);
                 serializer.serialize_str(string)
             }
             DataType::List(field) => {
-                let list_array: ListArray = self.value.clone().into();
-                let values = list_array.values();
-                let mut s = serializer.serialize_seq(Some(values.len()))?;
+                let list_array: &ListArray = self.value.as_list();
+                // let values: &UInt16Array = list_array.values().as_primitive();
```

Comment:
Right now we only support `ListArray`s for filling in ROS2 arrays. We could also support the `UInt8Array`/`Uint16Array`/etc types of arrow here. For this, we could use `as_list_opt`/` as_primitive_opt`/etc and choose the one that returns `Some`.

Reply:
`UInt8Array` is not a arrow DataType it's an array. The corresponding type is `UInt8`

Reply:
I meant the [`UInt8Array` type](https://docs.rs/arrow/latest/arrow/array/type.UInt8Array.html) of the `arrow` crate, which is a type alias for the [`PrimitiveArray`](https://docs.rs/arrow/latest/arrow/array/struct.PrimitiveArray.html) type.